src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Unix.cs

```diff
@@ -11,172 +11,331 @@
 
 namespace System.Net.Http
 {
-    public class HttpClientHandler : HttpMessageHandler
+    public partial class HttpClientHandler : HttpMessageHandler
     {
-        #region Properties
+        // Only one of these two handlers will be initialized.
+        private readonly CurlHandler _curlHandler;
+        private readonly ManagedHandler _managedHandler;
+        private readonly DiagnosticsHandler _diagnosticsHandler;
 
-        public virtual bool SupportsAutomaticDecompression => _curlHandler.SupportsAutomaticDecompression;
+        public HttpClientHandler()
+        {
+            if (UseManagedHandler)
+            {
+                _managedHandler = new ManagedHandler();
+                _diagnosticsHandler = new DiagnosticsHandler(_managedHandler);
+            }
+            else
+            {
+                _curlHandler = new CurlHandler();
+                _diagnosticsHandler = new DiagnosticsHandler(_curlHandler);
+            }
+        }
 
-        public virtual bool SupportsProxy => _curlHandler.SupportsProxy;
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                ((HttpMessageHandler)_curlHandler ?? _managedHandler).Dispose();
+            }
+            base.Dispose(disposing);
+        }
 
-        public virtual bool SupportsRedirectConfiguration => _curlHandler.SupportsRedirectConfiguration;
+        public virtual bool SupportsAutomaticDecompression => _curlHandler != null ?
+            _curlHandler.SupportsAutomaticDecompression :
+            _managedHandler.SupportsAutomaticDecompression;
```

Comment:
It is a common base type.  But none (or very few) of these members being accessed are defined on that base type.